integration_tests/valset_stress_test.go

```diff
@@ -0,0 +1,120 @@
+
+		startingNonce, err := s.getLastValsetNonce(gravityContract)
+		s.Require().NoError(err, "error getting starting nonce")
+
+		bondTokens := sdk.TokensFromConsensusPower(50000, sdk.DefaultPowerReduction)
+		bondCoin := sdk.NewCoin("testgb", bondTokens)
+
+		valAddress, err := s.chain.validators[3].keyInfo.GetAddress()
+		s.Require().NoError(err)
+		val := sdk.ValAddress(valAddress)
+
+		// Delegate about 5% of the total staking power.
+		s.Require().Eventuallyf(func() bool {
+			s.T().Logf("Sending in valset request (starting_eth_valset_nonce %d)", startingNonce)
+
+			s.T().Logf("Delegating %v to %v in order to generate a validator set update", bondCoin, orchAddress)
+
+			delegate := types.NewMsgDelegate(orchAddress, val, bondCoin)
+			response, err := s.chain.sendMsgs(*clientCtx, delegate)
+			if err != nil {
+				s.T().Logf("error: %s", err)
+				return false
+			}
+
+			if response.Code != 0 {
+				if response.Code != 32 {
+					s.T().Log(response)
+				}
+				return false
+			}
+			return true
+		}, 5*time.Minute, 10*time.Second, "Delegate to validator failed will retry")
+
+		// Verify that delegation went through.
+		s.T().Logf("verifying delegation")
+		s.Require().Eventuallyf(func() bool {
+
+			s.Require().NoError(err, "error querying delegator bonded validators")
+			queryClient := types.NewQueryClient(clientCtx)
+			res, err := queryClient.Delegation(context.Background(), &types.QueryDelegationRequest{DelegatorAddr: orchAddress.String(), ValidatorAddr: val.String()})
+			if err != nil {
+				s.T().Logf("error: %s", err)
+				return false
+			}
+			s.T().Logf("Here's the delegation response: %s", res.DelegationResponse)
+			return true
+		}, 20*time.Second, 1*time.Second, "Delegation wasn't successful")
+
+		// Query signer set, to make sure valset was updated.
+		s.T().Logf("verifying signerset")
+		s.Require().Eventuallyf(func() bool {
+
+			s.Require().NoError(err, "error querying signerset")
+			queryClient := gravity.NewQueryClient(clientCtx)
+			res, err := queryClient.LatestSignerSetTx(context.Background(), &gravity.LatestSignerSetTxRequest{})
+			if err != nil {
+				s.T().Logf("error: %s", err)
+				return false
+			}
+			s.T().Logf("Here's the last signerset response: %s", res.SignerSet)
+			return true
+		}, 20*time.Second, 1*time.Second, "Signerset can't be retrieved")
+
+		// Grab current nonce.
+		currentNonce, err := s.getLastValsetNonce(gravityContract)
+		s.Require().NoError(err, "error getting current nonce")
+
+		// Run a loop until you get a nonce higher than the initial nonce
+		s.Require().Eventuallyf(func() bool {
+			for currentNonce == startingNonce {
+				currentNonce, err = s.getLastValsetNonce(gravityContract)
+				if currentNonce != startingNonce {
+					return true
+				}
+			}
+			return true
+		}, 5*time.Minute, 10*time.Second, "Validator set is not yet updated")
+
+		if currentNonce != startingNonce {
+			s.T().Logf("Validator set successfully updated! nonce: %d", currentNonce)
```

Comment:
should this be asserted ? something like
`s.Truef(currentNonce != startingNonce, "Failed to update validator set")` ?